shaping/wrapping.go

```diff
@@ -1093,7 +1098,8 @@ func (l *LineWrapper) processBreakOption(option breakOption, config lineConfig)
 		// Reject invalid line break candidate and acquire a new one.
 		return breakInvalid, Output{}
 	}
-	candidateRun := cutRun(run, l.mapper.mapping, l.lineStartRune, option.breakAtRune)
+	isFirstInLine := len(l.scratch.alt) == 0
```

Comment:
My only concern here is that we're directly observing the `alt` buffer when the rest of the logic relies upon methods that hide the specifics of the buffering. Could we instead have a `candidateLen()` method that returned this quantity? It would pair nicely with the existing `candidateAppend()`.

Reply:
Sure thing. If I follow you, the implementation of `candidateLen()` would still be `return l.scratch.alt`; but it would be more robust if we later change the buffering implementation. Am I correct ?

Reply:
Yeah, that's precisely what I meant. :D

Reply:
Done in c07830e.